src/google/protobuf/generated_message_table_driven.cc

```diff
@@ -73,7 +73,7 @@ inline Arena* GetArena(MessageLite* msg, int64 arena_offset) {
 
 template <typename Type>
 inline Type* AddField(MessageLite* msg, int64 offset) {
-#if LANG_CXX11
+#if GOOGLE_HAVE_STD_IS_TRIVIALLY_ASSIGNABLE
```

Comment:
I think you need to keep LANG_CXX11? GOOGLE_HAVE_STD_IS_TRIVIALLY_ASSIGNABLE may be true even when C++11 is disabled (I don't see any check for C++11 there).